tools/releases/distros.sh

```diff
@@ -0,0 +1,186 @@
+BINTRAY_SUBJECT="kong"
+BINTRAY_REPOSITORY="kuma"
+
+
+function msg_green {
+  builtin echo -en "\033[1;32m"
+  echo "$@"
+  builtin echo -en "\033[0m"
+}
+
+
+function msg_red() {
+  builtin echo -en "\033[1;31m" >&2
+  echo "$@" >&2
+  builtin echo -en "\033[0m" >&2
+}
+
+
+function msg_err() {
+  msg_red $@
+  exit 1
+}
+
+
+function get_envoy() {
+  local distro=$1
+
+  local status=$(curl -L -o build/envoy-$distro -u $BINTRAY_USERNAME:$BINTRAY_API_KEY \
+    --write-out %{http_code} --silent --output /dev/null \
+    "https://kong.bintray.com/envoy/envoy-1.11.0-$distro")
+
+  [ "$status" -ne "200" ] && msg_err "Error: failed downloading Envoy"
+}
+
+
+function create_tarball {
+  local system=$1
+  local arch=$2
+  local distro=$3
+
+  local dest_dir=build/kuma-$system-$arch
+
+  rm -rf $dest_dir
+  mkdir $dest_dir
+  mkdir $dest_dir/bin
+  mkdir $dest_dir/conf
+
+  get_envoy $distro
+  chmod 755 build/envoy-$distro
+
+  cp -p build/envoy-$distro $dest_dir/bin/envoy
+  cp -p build/artifacts-$system-$arch/kuma-cp/kuma-cp $dest_dir/bin
+  cp -p build/artifacts-$system-$arch/kuma-dp/kuma-dp $dest_dir/bin
+  cp -p build/artifacts-$system-$arch/kumactl/kumactl $dest_dir/bin
+  cp -p build/artifacts-$system-$arch/kuma-tcp-echo/kuma-tcp-echo $dest_dir/bin
+  cp -p pkg/config/app/kuma-cp/kuma-cp.defaults.yaml $dest_dir/conf/kuma-cp.conf
+
+  cp tools/releases/templates/* $dest_dir
+
+  tar -czf build/artifacts-$system-$arch/kuma-$system-$arch.tar.gz -C $dest_dir .
+}
+
+
+function package {
+  # create binaries and tarballs with all binaries
+  for os in "${DISTRIBUTIONS[@]}"; do
+    local distro=$(echo "$os" | awk '{split($0,parts,":"); print parts[1]}')
+    local system=$(echo "$os" | awk '{split($0,parts,":"); print parts[2]}')
+
+    for arch in "${GOARCH[@]}"; do
+
+      msg_green "**** Packaging Kuma for $system-$arch... ****"
+      msg_green
+
+      make GOOS=$system GOARCH=$arch BUILD_INFO_GIT_TAG=$KUMA_VERSION BUILD_INFO_GIT_COMMIT=$KUMA_COMMIT build
+      create_tarball $system $arch $distro
+    done
+  done
+
+  msg_green
+  msg_green "**** Success, Kuma is built! ****"
+}
+
+
+function create_bintray_package {
+  local package_name=$1
+  local creation_status="$(curl --write-out %{http_code} --silent --output /dev/null \
+    -XPOST -H 'Content-Type: application/json' -u $BINTRAY_USERNAME:$BINTRAY_API_KEY\
+    -d '{"name":"'"$package_name"'"}' \
+    $BINTRAY_ENDPOINT/packages/$BINTRAY_SUBJECT/$BINTRAY_REPOSITORY)"
+  [ "$creation_status" -eq "409" ] && return
+  [ "$creation_status" -ne "201" ] && msg_err "Error: could not create package $package_name"
+}
+
+
+function release {
+  for os in "${DISTRIBUTIONS[@]}"; do
+    local distro=$(echo "$os" | awk '{split($0,parts,":"); print parts[1]}')
+    local system=$(echo "$os" | awk '{split($0,parts,":"); print parts[2]}')
+
+    for arch in "${GOARCH[@]}"; do
+      local artifact="build/artifacts-$system-$arch/kuma-$system-$arch.tar.gz"
```

Comment:
notice that artifact name doesn't include `distro`, which means that `centos` and `ubuntu` will be represented by the same `*.tar.gz` archive (which is not correct - `envoy` binary must differ)

Reply:
Fixed!